### ISSUE TITLE
Issue with viaroute example

### ISSUE THREAD

**Issue by Robinlovelace** [ropensci/stplanr#141]
This is what I get when running the viaroute example:

```r
 exroutes <- viaroute(viapoints=list(data.frame(x=c(-33.5,-33.6,-33.7),y=c(150,150.1,150.2))))
  r <- viaroute2sldf(exroutes)
```

```
Error in data.frame(routenum = i, routedesc = rep(x$routes$legs[[i]]$summary,  : 
  arguments imply differing number of rows: 1, 16, 8, 0
```

The other example works fine though. Any ideas @richardellison?

**Comment by richardellison** [comment]
I'll have a look, the error implies that the number of routes doesn't match the summary information which seems rather odd (and I'm sure this example worked when I tried it a few weeks ago).

**Comment by Robinlovelace** [comment]
Thanks for the fix - no longer errors but I get some strange warnings and the output looks strange:

```
> exroutes <- viaroute(viapoints=list(data.frame(x=c(-33.5,-33.6,-33.7),y=c(150,150.1,150.2))))
> r <- viaroute2sldf(exroutes)
Warning messages:
1: In bind_rows_(x, .id) : Unequal factor levels: coercing to character
2: In bind_rows_(x, .id) : Unequal factor levels: coercing to character
3: In bind_rows_(x, .id) : Unequal factor levels: coercing to character
4: In bind_rows_(x, .id) : Unequal factor levels: coercing to character
plot(r)
```

![screenshot from 2016-11-20 18-44-23](https://cloud.githubusercontent.com/assets/1825120/20465277/6ab726a6-af51-11e6-8a67-374b51c1f816.png)

**Comment by Robinlovelace** [comment]
Compared with this from the [**osrm** package](https://github.com/rCarto/osrm):

```r
install.packages("osrm")
library(osrm)
r_osrm <- osrmTrip(data.frame(i = 1:3,x=c(150,150.1,150.2),y=c(-33.5,-33.6,-33.7)))
plot(r_osrm[[1]]$trip)
```

![screenshot from 2016-11-20 18-54-45](https://cloud.githubusercontent.com/assets/1825120/20465342/d36cb70a-af52-11e6-9c02-91da2ddf6eb0.png)

**Comment by Robinlovelace** [comment]
Also this seems to generate an error with the **stplanr** implementation:

```r
origin = tmap::geocode_OSM("London")
destination = tmap::geocode_OSM("Leeds")
od_df = rbind(origin$coords, destination$coords)
exroutes <- viaroute(viapoints = list(od_df))
r <- viaroute2sldf(exroutes)
```

```
Error in data.frame(routenum = i, legnum = j, routedesc = rep(x$routes$legs[[i]]$summary[j],  : 
  arguments imply differing number of rows: 1, 2, 0
```

Interestingly this also seems to fail with **osrm**:

```r
ro = osrmRoute(origin, destination)
```

```
osrmRoute function returns an error: 
Error: URL string malformed close to position 31: "73, 51"
```

**Comment by Robinlovelace** [comment]
Update: it works with **osrm**: https://github.com/rCarto/osrm/issues/17

**Comment by richardellison** [comment]
Reopening, seeing the same thing. I'll have another look.

**Comment by Robinlovelace** [comment]
Any update on this @richardellison? Thinking better to fix bugs before adding new features and improving performance.

**Comment by richardellison** [comment]
This has been fixed in #161. The following now returns the correct geometries.

```{r}
exroutes <- viaroute(viapoints=list(data.frame(x=c(-33.5,-33.6,-33.7),y=c(150,150.1,150.2))))
r <- viaroute2sldf(exroutes)
```
![image](https://cloud.githubusercontent.com/assets/10625733/21050379/722d8fca-be6e-11e6-88ea-75b20ffb2236.png)

You will notice that this is different from the result returned by the osrm package. This is because the "osrm" package calculates the individual routes between each pair of coordinates while this implementation uses the coordinates as waypoints on a single route. To return the same result as the "osrm" package you can run the following:

```{r}
exroutes2 <- viaroute(viapoints=list(
    data.frame(x=c(-33.5,-33.7),y=c(150,150.2)),
    data.frame(x=c(-33.6,-33.7),y=c(150.1,150.2))
))
r2 <- viaroute2sldf(exroutes2)
```
![image](https://cloud.githubusercontent.com/assets/10625733/21050441/e4f61dce-be6e-11e6-8df1-f0757814b77d.png)
